plugins/inputs/influxdb_listener/http_listener.go

```diff
@@ -371,6 +388,16 @@ func (h *HTTPListener) parse(b []byte, t time.Time, precision string) error {
 	}
 
 	for _, m := range metrics {
+		// Do we need to keep the database name in the query string
+		if h.KeepDatabase {
+			// Did we get a database argument. If we didn't get it. We can't set it.
+			if db != "" {
+				// Is there already a database set. If not use the database in the query string.
+				if _, ok := m.Tags()[h.DatabaseTag]; !ok {
```

Comment:
Since the default will be off, it will be best to clobber the tag if it is already set.

Reply:
I'm not sure what you mean by this one, so I'm assuming you mean if the db is set but the tag is not we should just ignore it. That's what I did, so if you want to change that you can comment on the line again.